test/python/transpiler/test_clifford_passes.py

```diff
@@ -536,6 +537,122 @@ def test_do_not_merge_conditional_gates(self):
         # Make sure that the condition on the middle gate is not lost
         self.assertIsNotNone(qct.data[1].operation.condition)
 
+    def test_collect_with_cliffords(self):
+        """Make sure that collecting Clifford gates and replacing them by Clifford
+        works correctly when the gates include other cliffords."""
+
+        # Create a Clifford over 2 qubits
+        cliff_circuit = QuantumCircuit(2)
+        cliff_circuit.cx(0, 1)
+        cliff_circuit.h(0)
+        cliff = Clifford(cliff_circuit)
+
+        qc = QuantumCircuit(3)
+        qc.h(0)
+        qc.append(cliff, [1, 0])
+        qc.cx(1, 2)
+
+        # Collect clifford gates from the circuit (in this case all the gates must be collected).
+        qct = PassManager(CollectCliffords()).run(qc)
+        self.assertEqual(len(qct.data), 1)
+
+        # Make sure that the operator for the initial quantum circuit is equivalent to the
+        # operator for the collected clifford.
+        op1 = Operator(qc)
+        op2 = Operator(qct)
+        self.assertTrue(op1.equiv(op2))
```

Comment:
These tests use `equiv` rather than `==` - the former normalises global phase, so `==` is a strong condition.  Do we need the global-phase fuzziness here?  If so, it seems a bit unnerving, since the transpiler is supposed to maintain the global phase entirely.

(Comment repeats for all the tests here.)

Reply:
Ouch, your comment points to a real problem with `OptimizeCliffords` transpiler pass, that I should have but did not think about. 

The problem is that Cliffords are only defined up to a global phase: while "in the real world" `XY = iZ`, when using Clifford's stabilizer/destabilizer tableaus, the multiplication by `i` get dropped, i.e. "in the Clifford world" `XY=Z`. This is also the reason why most of the tests that construct an `Operator` out of `Clifford` in `test_clifford.py` check for `equiv` and not for `==`. 

The above in code:
```
qc = QuantumCircuit(1)
qc.x(0)
qc.y(0)
print(Operator(Clifford(qc)) == Operator(qc))
```
print the value `False`.

So right now replacing a block of Clifford gates by a Clifford may drop the global phase, which is indeed the problem to "the transpiler maintaining the global phase". That is, the `OptimizeCliffords` transpiler pass can only guarantee equivalence up to a global phase.

Note: I do believe that when optimizing Cliffords over _a subset of qubits_ the problem is still only with _global_ and not _local_ phases.

BTW, I thought there was an effort to incorporate the full phase into the `Clifford` class, maybe @ShellyGarion or @ikkoham can comment? This would immediately solve our problems here.

Reply:
So I don't think we should replace `equiv` by `==` in these tests. However, we should see if there is already a solution for maintaining the global phase. And if not we should clearly document this caveat in the documentation for `OptimizeCliffords` transpiler pass. What do you think?

Reply:
That sounds like a sensible plan to me. If the phase non-equivalence is already in the pass and in Terra 0.22, then there's no massive rush to fix it now before the release, and anyway, we can treat it as a bug fix. Let's open a new issue and discuss the right way forwards.